src/components/EditableHeading/EditableHeading.tsx

```diff
@@ -25,143 +18,34 @@ export interface EditableHeadingProps extends VibeComponentProps {
-
-    const [isEditing, setIsEditing] = useState(false);
-    const [inputValue, setInputValue] = useState(value);
-    const [inputWidth, setInputWidth] = useState(0);
-
-    const inputRef = useRef(null);
-
-    function toggleEditMode() {
-      if (readOnly || isEditing) {
-        return;
-      }
-
-      setIsEditing(true);
-    }
-
-    function handleInputValueChange() {
-      setIsEditing(false);
-      if (!inputValue || value === inputValue) {
-        setInputValue(value);
-        return;
-      }
-      onChange?.(inputValue);
-    }
-
-    function handleBlur() {
-      handleInputValueChange();
-    }
-
-    function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
-      if (event.key === keyCodes.ENTER) {
-        handleInputValueChange();
-      }
-      if (event.key === keyCodes.ESCAPE) {
-        setIsEditing(false);
-        setInputValue(value);
-      }
-    }
-
-    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
-      setInputValue(event.target.value);
-    }
-
-    const clickableProps = useClickableProps(
-      {
-        onClick: toggleEditMode,
-        id,
-        role: "button"
-      },
-      mergedRef
-    );
-
-    function focus() {
-      if (inputRef.current) {
-        inputRef.current?.focus();
-      }
-    }
-
-    useEffect(() => {
-      if (isEditing) {
-        focus();
-      }
-    }, [isEditing]);
-
-    const inputClassNames = useMemo(() => {
-      return cx(styles.input, getStyle(styles, camelCase(type + "-" + weight)));
-    }, [type, weight]);
-
     return (
-      <div
-        ref={mergedRef}
-        id={id}
-        aria-label={ariaLabel}
+      <EditableTypography
+        ref={ref}
         data-testid={dataTestId || getTestId(ComponentDefaultTestId.EDITABLE_HEADING, id)}
-        className={cx(styles.editableHeading, className)}
-      >
-        {isEditing && !readOnly ? (
-          <>
-            <HiddenInputPlaceholder
-              className={inputClassNames}
-              value={inputValue || placeholder}
-              onChange={setInputWidth}
-            />
-            <input
-              ref={inputRef}
-              className={inputClassNames}
-              value={inputValue}
-              onChange={handleChange}
-              onKeyDown={handleKeyDown}
-              onBlur={handleBlur}
-              aria-label={ariaLabel}
-              placeholder={placeholder}
-              style={{ width: inputWidth }}
-              role="input"
-            />
-          </>
-        ) : (
-          <Heading
-            {...clickableProps}
-            className={cx(styles.heading, { [styles.disabled]: readOnly })}
-            type={type}
-            weight={weight}
-          >
-            {inputValue}
+        renderTypography={({ value, className, ...clickableProps }) => (
+          <Heading type={type} weight={weight} className={className} {...clickableProps}>
+            {value}
```

Comment:
I was wondering about a couple of ways of composing `EditableText` and `EditableTypography` out of the generic `EditableTypography`, using render props seems most reasonable, this will allow us (when needed) to easily make any typography component editable. Other alternatives might be providing a component as a prop (but this means prop drilling - context will work but it will be less explicit), or passing the typography component as children which will probably require using `cloneElement` to inject additional props but [it's not recommended now](https://react.dev/reference/react/cloneElement)